front/lib/api/data_sources.ts

```diff
@@ -167,14 +168,25 @@ export async function getDataSources(
  */
 export async function softDeleteDataSourceAndLaunchScrubWorkflow(
   auth: Authenticator,
-  dataSource: DataSourceResource,
-  transaction?: Transaction
+  {
+    dataSource,
+    transaction,
+  }: {
+    dataSource: DataSourceResource;
+    transaction?: Transaction;
+  }
 ): Promise<
   Result<DataSourceType, { code: "unauthorized_deletion"; message: string }>
 > {
   const owner = auth.getNonNullableWorkspace();
-
-  if (!auth.isBuilder()) {
+  const isAuthorized =
+    dataSource.space.canWrite(auth) ||
+    // Only allow to remote database connectors if the user is an admin.
+    (dataSource.space.isSystem() &&
+      dataSource.space.canAdministrate(auth) &&
```

Comment:
canAdministrate should be simply always allowed no?

Reply:
Sorry my point was that if you `canAdministrate` I thiiiiink you should be always allowed to delete a data source?

Right now it's not the case based on the above since it's restricted to `canWrite or (isSystem and canAdministrate and isRemote)`. I would argue we could simplify to `canWrite or canAdministrate` ?